BaikalDataGenerator/Source/config_loader.cpp

```diff
@@ -0,0 +1,226 @@
+/**********************************************************************
+Copyright (c) 2018 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#include "config_loader.h"
+#include "XML/tinyxml2.h"
+#include "material_io.h"
+#include <filesystem>
+
+// validation checks helper macroses to reduce copy paste
+#define ASSERT_PATH(file_name, message) \
+     if (file_name.empty()) { \
+         THROW_EX(message) } \
+
+#define ASSERT_EXT(file_name, message) \
+     if (camera_file.extension() != ".xml") { \
+         THROW_EX(message) } \
+
+#define ASSERT_FILE_EXISTING(file_name, message) \
+     if (!std::filesystem::exists(file_name)) { \
+         THROW_EX(message) } \
+
+void ConfigLoader::ValidateConfig(const DGenConfig& config) const
+{
+    auto camera_file = config.camera_file;
+    auto light_file = config.light_file;
+    auto spp_file = config.spp_file;
+    auto scene_file = config.scene_file;
+    auto output_dir = config.output_dir;
+
+    // validate input config
+    ASSERT_PATH(camera_file, "missed camera file");
```

Comment:
This macro may take a single paramter and form the whole message automatically.
`ASSERT_PATH(camera_file)` -> `File camera_file not found: /some/path`.